Various Azure IoT Hub doc improvements (#6238)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * Added handling of MQTT ping failures and MQTT input failures.
  * Updated the API version used in MQTT connection to Azure IoT Hub to 2020-09-30.
  * Added the :c:func:`azure_iot_hub_dps_reset` function for resetting the DPS information.

>   * Added information in the documenation about which credentials to use and where to find them.

Reviewer: ```suggestion
  * Added a note about the credentials and their location.
```

Author: Fixed
